packages/playwright-core/src/server/page.ts

```diff
@@ -227,6 +227,32 @@ export class Page extends SdkObject {
     this._browserContext.emit(event, ...args);
   }
 
+  async resetForReuse(metadata: CallMetadata) {
+    this.setDefaultNavigationTimeout(undefined);
+    this.setDefaultTimeout(undefined);
+
+    await this.removeExposedBindings();
+    await this.removeInitScripts();
+
+    // TODO: handle pending routes.
+    await this._frameManager.closeOpenDialogs();
```

Comment:
Let's do this before `removeExposedBindings()`, otherwise the latter might stall because renderer is stuck?